### ISSUE TITLE
Spring4shell vulnerability

### ISSUE THREAD

**Issue by dkemens** [Overv/openstreetmap-tile-server#269]
Hello!

This amazing package is currently being flagged vulnerable for the [spring4shell exploit](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2022-22965):

## Scan Result
{
  "result": "FAIL",
  "hostname": "1b97dd72169d",
  "scan_ts": "2022-04-13T15:48:43+0000",
  "scan_v": "0.3",
  "search_root": ".",
  "vulnerable_application_paths": [ "./usr/share/java/spring3-beans-3.x.jar" ],
  "unknown_application_paths": [],
  "java9plus_paths": [ "./usr/lib/jvm/java-11-openjdk-amd64/bin/java" ],
  "unknown_java_paths": [ "./var/lib/dpkg/alternatives/java" ]
}

I looked through the repository, but it wasn't clear to me where the java/spring dependency is coming from. I'm hoping a maintainer might be able to quickly direct me to the source of those packages so I can work on expediting on a patch.

Thanks!

**Comment by Istador** [comment]
According to the build logs, spring should be installed at this step:
https://github.com/Overv/openstreetmap-tile-server/blob/cffbd7be5b14f060c4e4d4c83f7cf0bdd55c6541/Dockerfile#L110-L140

Specifically it installs:
```
#10 18.62 Get:487 http://archive.ubuntu.com/ubuntu focal/universe amd64 libspring-core-java all 4.3.22-4 [1024 kB]
#10 18.67 Get:488 http://archive.ubuntu.com/ubuntu focal/universe amd64 libspring-beans-java all 4.3.22-4 [669 kB]
#10 18.69 Get:489 http://archive.ubuntu.com/ubuntu focal/universe amd64 libspring-aop-java all 4.3.22-4 [376 kB]
#10 18.70 Get:490 http://archive.ubuntu.com/ubuntu focal/universe amd64 libspring-expression-java all 4.3.22-4 [252 kB]
#10 18.71 Get:491 http://archive.ubuntu.com/ubuntu focal/universe amd64 libspring-context-java all 4.3.22-4 [973 kB]
#10 18.73 Get:492 http://archive.ubuntu.com/ubuntu focal/universe amd64 libspring-transaction-java all 4.3.22-4 [221 kB]
#10 18.74 Get:493 http://archive.ubuntu.com/ubuntu focal/universe amd64 libspring-jdbc-java all 4.3.22-4 [376 kB]
```

If I see it correctly, it is a dependency of `osmosis`.

`osmosis` is used during the automated updates and does download/parse external data from the internet, so in theory it might be a possible injection path.

There is no `tomcat` involved, so according to the CVE descruption it shouldn't be vulnerable?

Do you know if it is already fixed for the `ubuntu:20.04` base image?

**Comment by dkemens** [comment]
This is very helpful, thank you. Looks like the last osmosis release was in 2020, so this definitely isn't on their radar. However, if Tomcat isn't used, then the vulnerability can't be exploited - so perhaps it's a mum issue. Thanks again for directing me towards the culprit, perhaps I'll hit up the OSM community board and see if there's a consensus.

Thanks!
